pkg/customresourcestate/registry_factory.go

```diff
@@ -217,7 +217,41 @@ func (c *compiledGauge) Values(v interface{}) (result []eachValue, errs []error)
 	switch iter := v.(type) {
 	case map[string]interface{}:
 		for key, it := range iter {
-			ev, err := c.value(it)
+			// TODO: Handle multi-length valueFrom paths (https://github.com/kubernetes/kube-state-metrics/pull/1958#discussion_r1099243161).
+			// Try to deduce `valueFrom`'s value from the current element.
+			var ev *eachValue
+			var err error
+			var didResolveValueFrom bool
+			// `valueFrom` will ultimately be rendered into a string and sent to the fallback in place, which also expects a string.
+			// So we'll do the same and operate on the string representation of `valueFrom`'s value.
+			sValueFrom := c.ValueFrom.String()
+			// No comma means we're looking at a unit-length path (in an array).
+			if !strings.Contains(sValueFrom, ",") &&
+				sValueFrom[0] == '[' && sValueFrom[len(sValueFrom)-1] == ']' &&
+				// "[...]" and not "[]".
+				len(sValueFrom) > 2 {
+				extractedValueFrom := sValueFrom[1 : len(sValueFrom)-1]
```

Comment:
This is a bit odd to me. ValueFrom is orginally a string array and here we are manually trying to parse it as a string and extracting the first element from it, when we should have an easy access to it. What do you think about adding a new method to `valuePath` that would return a string array instead of a raw string?

Reply:
@rexagod I am merging the PR without that change since it is mostly code quality related, but if my suggestion makes sense to you, let's address that in a follow-up